docs/content/customization/generic/dns.md

```diff
@@ -48,3 +48,32 @@ Applying this configuration will result in the following value being set:
             imageRepository: "my-registry.io/my-org/my-repo"
             imageTag: "v1.11.3_custom.0"
     ```
+
+The CoreDNS version can also be updated automatically. To do this, set `coreDNS` to an empty object:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: <NAME>
+spec:
+  topology:
+    variables:
+      - name: clusterConfig
+        value:
+          dns:
+            coreDNS: {}
+```
+
+Applying this configuration will result in the following value being set,
+with the version of the CoreDNS image being set based on the cluster's Kubernetes version:
+
+- `KubeadmControlPlaneTemplate`:
+
+  - ```yaml
+    spec:
+      kubeadmConfigSpec:
+        clusterConfiguration:
+          dns:
+            imageTag: "v1.11.3"
+    ```
```

Comment:
As this should be the default and preferred behaviour, could you move this above the explicit version setting please.

Reply:
Done in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/959/commits/b47e1b792e9d0ec5173b8c25e4e47646b89733bb